telemetry-impls/clean-up-artifacts/action.yml

```
@@ -0,0 +1,52 @@
+name: 'clean-up-artifacts'
+description: |
+  Artifact cleanup removes all of the temporary "carrier" files that are created
+  by each build job to transmit attribute values. Otherwise, these would be
+  noisy and make it harder to find "real," interesting artifacts.
+
+runs:
+  using: 'composite'
+  steps:
+    - name: Clean up telemetry files
+      uses: actions/github-script@v7
+      if: runner.debug != '1'
+      with:
+        retries: 3
+        script: |
+          const runAttempt = parseInt(process.env.GITHUB_RUN_ATTEMPT, 10)
+          get_artifacts = async ({github, context, process}) => {
+            const opts = github.rest.actions.listWorkflowRunArtifacts.endpoint.merge(
+                {
+                    attempt_number: runAttempt,
+                    owner: context.repo.owner,
+                    repo: context.repo.repo,
+                    run_id: context.runId,
+                    per_page: 100
+                });
+            const artifacts = await github.paginate(opts);
+            return artifacts;
+          };
+
+          var artifacts = undefined;
+          var retryCount = 0;
+          maxRetries = 5;
+          initialDelay = 2000;
+          while (artifacts === undefined) {
+            artifacts = await get_artifacts({github, context, process});
+            if (artifacts === undefined && retryCount < maxRetries) {
+              retryCount++;
+              await new Promise((res) => setTimeout(res, initialDelay * Math.pow(2, retryCount - 1)));
+            } else {
+              break;
+            }
+          }
+
+          artifacts.forEach(artifact => {
+            if (artifact.name.startsWith('telemetry-tools-')) {
+              github.rest.actions.deleteArtifact({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                artifact_id: artifact.id
+              });
+            }
+          });
```

```suggestion
          await Promise.all(
            artifacts
            .filter(artifact => artifact.name.startsWith('telemetry-tools-'))
            .map(artifact => {
              github.rest.actions.deleteArtifact({
                owner: context.repo.owner,
                repo: context.repo.repo,
                artifact_id: artifact.id
              });
            })
          );
```

The `github.rest.actions.deleteArtifact` function returns a promise that needs to be awaited. This suggestion splits the logic into a `.filter` and `.map` chain which will effectively return an array of promises that can all be awaited with `await Promise.all()`.